### ISSUE TITLE
Overlap in IQR info box with stringTests

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/center-and-variability#540]
**Test device**
MacBook Air M1 chip

**Operating System**
13.5.1

**Browser**
Safari 16.6

**Problem description**
For https://github.com/phetsims/qa/issues/976, with stringTest=dynamic (double, long, or rtl), the text overlaps in the IQR checkbox after balls have been kicked. 

**Steps to reproduce**
1. Add ?stringTest=dynamic to the end of the url
2. On the Variability screen, press the Kick 5 button
3. Switch to the IQR panel
4. Press the info button
5. Press the right arrow 

**Visuals**
<img width="380" alt="Screenshot 2023-09-07 at 3 58 22 PM" src="https://github.com/phetsims/center-and-variability/assets/87318828/2489cdab-dfb9-4036-8f73-195b1f110ebd">

Also seen after pressing the down arrow a few times:
<img width="314" alt="Screenshot 2023-09-07 at 3 58 07 PM" src="https://github.com/phetsims/center-and-variability/assets/87318828/326c4526-bfc4-4651-bf51-f4e19bcd4f4a">

**Comment by marlitas** [comment]
This one tested me, and was pretty fun to figure out. Over to @matthew-blackman for code review. @Nancy-Salpepi can you check that all looks well on main?

**Comment by Nancy-Salpepi** [comment]
This is looking much better @marlitas! The only thing is that the highlighted are doesn't expand with the text. 
<img width="323" alt="Screenshot 2023-09-12 at 4 16 37 PM" src="https://github.com/phetsims/center-and-variability/assets/87318828/5e458c14-9c78-4115-be30-250b24334c1a">

**Comment by matthew-blackman** [comment]
Gonna have a go at recomputing the highlight rectangles and re-centering the text.

**Comment by matthew-blackman** [comment]
This is ready for both design and dev review! The latest commit updates the highlight rectangles around the Q1 and Q3 labels, and also keeps the labels centered.

**Comment by Nancy-Salpepi** [comment]
Highlighted area is looking really nice.
There is something odd that I missed yesterday 😞 that a very small ball distribution pointed out to me today....

<img width="237" alt="Screenshot 2023-09-13 at 2 38 50 PM" src="https://github.com/phetsims/center-and-variability/assets/87318828/2e8f541c-4318-4b57-ab16-b11c07d48106">

**Comment by marlitas** [comment]
> There is something odd that I missed yesterday

I don't think you missed it @Nancy-Salpepi because I actually remember running into this and calling the full update function to rearrange the plot node instead of just the function that handled label overlap, and it took care of this situation. I imagine that got missed in this latest refactor since it's such a small detail. I haven't had a chance to look at implementation yet for the new code @matthew-blackman, but I can wait until this bug is resolved?

**Comment by matthew-blackman** [comment]
Good find @Nancy-Salpepi ! Sorry I missed this. Fixed in the above commit, please confirm 🙂

@marlitas I also did my best to simplify the multilink by pulling out updateIntervalBar(). Looking forward to your review!

**Comment by Nancy-Salpepi** [comment]
All looks good now on my end!

**Comment by marlitas** [comment]
The changes look great. Thanks for factoring all that out @matthew-blackman! There was a small grammar correction I committed and then one readability thing that might just be me.

I don't know if the `forEach` is necessary here. I would just add each child individually. It makes it harder to skim and I had to stop a second longer to figure out we're just adding children. I mean it obviously works and isn't a efficiency problem, more just a readability thing for me, but I'm pretty indifferent on wether you change it or not. Just thought I'd bring it up.

In `IQRNode.ts`:
```js
nonMedianLabelNodes.forEach( nonMedianLabelNode => {
      boxWhiskerNode.addChild( nonMedianLabelNode );
    } );
 boxWhiskerNode.addChild( medianLabelNode );
```
